src/Interpolators.f90

```diff
@@ -0,0 +1,146 @@
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!                          Futility Development Group                          !
+!                             All rights reserved.                             !
+!                                                                              !
+! Futility is a jointly-maintained, open-source project between the University !
+! of Michigan and Oak Ridge National Laboratory.  The copyright and license    !
+! can be found in LICENSE.txt in the head directory of this repository.        !
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!> @brief A Fortran 2003 module implementing interpolation routines for input
+!>        arrays returning either interpolated values or a flat "extrapolation"
+!>        when interpolation is out of range.
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+MODULE Interpolators
+
+USE IntrType
+IMPLICIT NONE
+
+PUBLIC :: Interp
+
+!> @brief Generic interface to interpolate whatever is handed in.
+!>
+INTERFACE Interp
+  !> @copybrief Interpolators::Interp_1D
+  !> @copydetails Interpolators::Interp_1D
+  MODULE PROCEDURE Interp_1D
+  !> @copybrief Interpolators::Interp_2D
+  !> @copydetails Interpolators::Interp_2D
+  MODULE PROCEDURE Interp_2D
+  !> @copybrief Interpolators::Interp_3D
+  !> @copydetails Interpolators::Interp_3D
+  MODULE PROCEDURE Interp_3D
+ENDINTERFACE Interp
+!
+!===============================================================================
+CONTAINS
+!
+!-------------------------------------------------------------------------------
+!> @brief This routine takes a 1D data "table" and linearly interpolates to find
+!>        a desired point.input arrays and interpolates as needed. It is important
```

Comment:
got some typo action going on here and in the other 2 `Interp_*` routine headers